File: src/low_med.py
import hashlib
import random
import tempfile


# MEDIUM: Insecure Randomness
def generate_discount_code():
    # Rule: py/insecure-randomness
    code = random.randint(1000, 9999)
    return str(code)


# LOW: Insecure Temporary File & Weak Cryptography
def create_temp_hash(data):
    # Rule: py/weak-crypto
    weak_hash = hashlib.sha1(data.encode()).hexdigest()

    # Rule: py/insecure-temporary-file
    tmp = tempfile.mktemp()  # noqa: F841
    return weak_hash


# LOW: Empty Except Block
def parse_data(raw_data):
    try:
        parsed = int(raw_data)
        return parsed
    except Exception:  # noqa: E722
        # Rule: py/empty-except
        pass
    return 0


if __name__ == "__main__":
    print(generate_discount_code())
    print(create_temp_hash("test_data"))
    print(parse_data("abc"))

File: js/low_med.js
const crypto = require('crypto');

// MEDIUM: Insecure Randomness
function createDiscountCode() {
    // Rule: js/insecure-randomness
    return Math.floor(Math.random() * 10000);
}

// LOW: Weak Cryptographic Algorithm
function generateHash(data) {
    if (!data) {
        data = "default_data";
    }
    // Rule: js/weak-cryptographic-algorithm
    return crypto.createHash('sha1').update(data).digest('hex');
}

console.log(createDiscountCode());
console.log(generateHash("test"));
